### DISCUSSION TITLE
New option: restrict canvas to horizontal/vertical mode

### DISCUSSION THREAD

**Discussion by Sporknife** [flxzt/rnote#893] [Ideas]
Currently, the canvas can expand infinitely in vertical and horizontal direction. I suggest adding an option to restrict it to horizontal/vertical mode only. This could also reduce the amount of memory the app uses if the user restricts it to only one direction because you don't have to render that much anymore.

The user could change the mode of the canvas direction while already having a something drawn, but this is prone to problems like wrong page layouting (the page on the left/right might be somewhat connected to the page of opposite direction) so this would probably require some checks. I envisioned this to something like this:

1. Are all used pages in a single column? 
 - Yes? Done!
 - No?   GOTO: 2. Would you like to mer...  OR  GOTO: 3. Showcase pages...
2. Would you like to merge pages into one? (if possible)
 - Yes? Select which ones.   GOTO: 1. Are all used...
 - No?   GOTO: 3. Showcase pages...
3. Showcase pages in the selected alignment. Allow the user to change the position of the page in the selected mode.

**Comment by flxzt** [comment]
Are you aware of the different document layouts? 
![grafik](https://github.com/flxzt/rnote/assets/19841886/a36f9157-4143-4ab2-b8eb-cc7b62da24fa)
![grafik](https://github.com/flxzt/rnote/assets/19841886/a009f291-1c77-4cea-9646-dc09bc54bfdb)

**Reply by Sporknife** [reply]
Let me dig a hole and bury myself, please.
